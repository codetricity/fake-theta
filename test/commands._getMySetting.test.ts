import {
  request,
  toMatchInvalidHeaderParameterError,
  toMatchMissingParameterError,
} from './helper';

describe('POST /osc/commands/execute _getMySetting', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._getMySetting',
        parameters: { mode: 'image' },
      })
      .expect(200);
    expect(res.body).toMatchSnapshot();
  });

  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._getMySetting',
        parameters: { mode: 'video' },
      })
      .expect(200);
    expect(res.body).toMatchSnapshot();
  });

  it('should respond missingParameterError', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._getMySetting',
      })
      .expect(400);
    toMatchMissingParameterError(res);
  });
});

describe('THETA X POST /osc/commands/execute _getMySetting', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._getMySetting',
        parameters: { mode: 'image' },
      })
      .set('emulating-theta-model', 'x')
      .expect(200);
    expect(res.body).toMatchSnapshot();
  });

  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._getMySetting',
        parameters: { mode: 'video' },
      })
      .set('emulating-theta-model', 'x')
      .expect(200);
    expect(res.body).toMatchSnapshot();
  });

  it('should respond missingParameterError', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._getMySetting',
      })
      .set('emulating-theta-model', 'x')
      .expect(400);
    toMatchMissingParameterError(res);
  });
});

describe('THETA Z1 POST /osc/commands/execute _getMySetting', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._getMySetting',
        parameters: { mode: 'image' },
      })
      .set('emulating-theta-model', 'z1')
      .expect(200);
    expect(res.body).toMatchSnapshot();
  });

  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._getMySetting',
        parameters: { mode: 'video' },
      })
      .set('emulating-theta-model', 'z1')
      .expect(200);
    expect(res.body).toMatchSnapshot();
  });

  it('should respond missingParameterError', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._getMySetting',
      })
      .set('emulating-theta-model', 'z1')
      .expect(400);
    toMatchMissingParameterError(res);
  });
});

describe('Unknown Type POST /osc/commands/execute _getMySetting', () => {
  it('should respond invalidHeaderParameterError', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._getMySetting',
      })
      .set('emulating-theta-model', 'unknown')
      .expect(400);
    toMatchInvalidHeaderParameterError(res);
  });
});
