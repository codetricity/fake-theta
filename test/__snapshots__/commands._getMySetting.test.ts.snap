// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST /osc/commands/execute _getMySetting should respond with results 1`] = `
{
  "_aiAutoThumbnail": "OFF",
  "_autoBracket": {
    "_bracketNumber": 2,
    "_bracketParameters": [
      {
        "_colorTemperature": 5000,
        "exposureCompensation": 0,
        "exposureProgram": 1,
        "iso": 400,
        "shutterSpeed": 0.004,
      },
      {
        "_colorTemperature": 5000,
        "exposureCompensation": 0,
        "exposureProgram": 1,
        "iso": 400,
        "shutterSpeed": 0.004,
      },
    ],
  },
  "_colorTemperature": 5000,
  "_faceDetect": "OFF",
  "_filter": "off",
  "_shootingMethod": "normal",
  "_timeShift": {
    "firstInterval": 2,
    "firstShooting": "front",
    "secondInterval": 5,
  },
  "_waterHousing": "OFF",
  "_waterHousingStitching": "underwater",
  "captureInterval": 6,
  "captureNumber": 0,
  "exposureCompensation": 0,
  "exposureDelay": 0,
  "exposureProgram": 2,
  "fileFormat": {
    "_mode": {},
    "height": 2752,
    "type": "jpeg",
    "width": 5504,
  },
  "iso": 400,
  "isoAutoHighLimit": 1600,
  "shutterSpeed": 0.004,
  "whiteBalance": "auto",
}
`;

exports[`POST /osc/commands/execute _getMySetting should respond with results 2`] = `
{
  "_bitrate": "Normal",
  "_colorTemperature": 5000,
  "_gain": "normal",
  "_gainExternal": "5",
  "_maxRecordableTime": 300,
  "_topBottomCorrection": "ApplyAuto",
  "_waterHousing": "OFF",
  "_waterHousingStitching": "underwater",
  "exposureCompensation": 0,
  "exposureDelay": 0,
  "exposureProgram": 2,
  "fileFormat": {
    "_codec": "H.264/MPEG-4 AVC",
    "_frameRate": 30,
    "height": 1920,
    "type": "mp4",
    "width": 3840,
  },
  "iso": 400,
  "isoAutoHighLimit": 1600,
  "shutterSpeed": 0.004,
  "whiteBalance": "auto",
}
`;

exports[`THETA X POST /osc/commands/execute _getMySetting should respond with results 1`] = `
{
  "_aiAutoThumbnail": "OFF",
  "_autoBracket": {
    "_bracketNumber": 2,
    "_bracketParameters": [
      {
        "_colorTemperature": 5000,
        "exposureCompensation": 0,
        "exposureProgram": 1,
        "iso": 400,
        "shutterSpeed": 0.004,
      },
      {
        "_colorTemperature": 5000,
        "exposureCompensation": 0,
        "exposureProgram": 1,
        "iso": 400,
        "shutterSpeed": 0.004,
      },
    ],
  },
  "_colorTemperature": 5000,
  "_faceDetect": "OFF",
  "_filter": "off",
  "_shootingMethod": "normal",
  "_timeShift": {
    "firstInterval": 2,
    "firstShooting": "front",
    "secondInterval": 5,
  },
  "_waterHousing": "OFF",
  "_waterHousingStitching": "underwater",
  "captureInterval": 6,
  "captureNumber": 0,
  "exposureCompensation": 0,
  "exposureDelay": 0,
  "exposureProgram": 2,
  "fileFormat": {
    "_mode": {},
    "height": 2752,
    "type": "jpeg",
    "width": 5504,
  },
  "iso": 400,
  "isoAutoHighLimit": 1600,
  "shutterSpeed": 0.004,
  "whiteBalance": "auto",
}
`;

exports[`THETA X POST /osc/commands/execute _getMySetting should respond with results 2`] = `
{
  "_bitrate": "Normal",
  "_colorTemperature": 5000,
  "_gain": "normal",
  "_gainExternal": "5",
  "_maxRecordableTime": 300,
  "_topBottomCorrection": "ApplyAuto",
  "_waterHousing": "OFF",
  "_waterHousingStitching": "underwater",
  "exposureCompensation": 0,
  "exposureDelay": 0,
  "exposureProgram": 2,
  "fileFormat": {
    "_codec": "H.264/MPEG-4 AVC",
    "_frameRate": 30,
    "height": 1920,
    "type": "mp4",
    "width": 3840,
  },
  "iso": 400,
  "isoAutoHighLimit": 1600,
  "shutterSpeed": 0.004,
  "whiteBalance": "auto",
}
`;

exports[`THETA Z1 POST /osc/commands/execute _getMySetting should respond with results 1`] = `
{
  "_autoBracket": {
    "_bracketNumber": 2,
    "_bracketParameters": [
      {
        "_colorTemperature": 5000,
        "aperture": 2.1,
        "exposureCompensation": 0,
        "exposureProgram": 1,
        "iso": 400,
        "shutterSpeed": 0.004,
        "whiteBalance": "auto",
      },
      {
        "_colorTemperature": 5000,
        "aperture": 2.1,
        "exposureCompensation": 0,
        "exposureProgram": 1,
        "iso": 400,
        "shutterSpeed": 0.004,
        "whiteBalance": "auto",
      },
    ],
  },
  "_burstOption": {
    "_burstBracketStep": 0,
    "_burstCaptureNum": 1,
    "_burstCompensation": 0,
    "_burstEnableIsoControl": 0,
    "_burstMaxExposureTime": 15,
    "_burstOrder": 0,
  },
  "_colorTemperature": 5000,
  "_compositeShootingOutputInterval": 600,
  "_compositeShootingTime": 86400,
  "_filter": "off",
  "_shootingMethod": "normal",
  "_whiteBalanceAutoStrength": "OFF",
  "aperture": 0,
  "captureInterval": 6,
  "captureNumber": 0,
  "exposureCompensation": 0,
  "exposureDelay": 0,
  "exposureProgram": 2,
  "fileFormat": {
    "height": 3360,
    "type": "jpeg",
    "width": 6720,
  },
  "iso": 0,
  "isoAutoHighLimit": 6400,
  "shutterSpeed": 0,
  "whiteBalance": "auto",
}
`;

exports[`THETA Z1 POST /osc/commands/execute _getMySetting should respond with results 2`] = `
{
  "_colorTemperature": 5000,
  "_filter": "off",
  "_maxRecordableTime": 300,
  "_whiteBalanceAutoStrength": "OFF",
  "aperture": 0,
  "exposureCompensation": 0,
  "exposureDelay": 0,
  "exposureProgram": 2,
  "fileFormat": {
    "_codec": "H.264/MPEG-4 AVC",
    "height": 1920,
    "type": "mp4",
    "width": 3840,
  },
  "iso": 0,
  "isoAutoHighLimit": 6400,
  "shutterSpeed": 0,
  "whiteBalance": "auto",
}
`;
